Database/MySQL/PreparedStatement.hs

```diff
@@ -0,0 +1,485 @@
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.Date
+                    , mysqlBindBuffer = castPtr ptr
+                    }
+
+        Time t ->
+            let (seconds, secondParts) = picoToSecondsAndMicroseconds (todSec t)
+                myTime = MYSQL_TIME
+                    { mysqlTimeYear = 0
+                    , mysqlTimeMonth = 0
+                    , mysqlTimeDay = 0
+                    , mysqlTimeHour = fromIntegral $ todHour t
+                    , mysqlTimeMinute = fromIntegral $ todMin t
+                    , mysqlTimeSecond = fromInteger seconds
+                    , mysqlTimeNeg = False
+                    , mysqlTimeSecondPart = fromInteger secondParts
+                    }
+            in
+            with myTime $ \ptr ->
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.Time
+                    , mysqlBindBuffer = castPtr ptr
+                    }
+        DateTime dt ->
+            with (localTime dt) $ \ptr ->
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.DateTime
+                    , mysqlBindBuffer = castPtr ptr
+                    }
+
+        Timestamp ts ->
+            with (localTime ts) $ \ptr ->
+                k $ defaultBind
+                    { mysqlBindBufferType = Types.Timestamp
+                    , mysqlBindBuffer = castPtr ptr
+                    }
+
+  where
+      localTime t =
+        let (year, month, dayOfMonth) = toGregorian $ localDay t
+            tod = localTimeOfDay t
+            (seconds, secondParts) = picoToSecondsAndMicroseconds (todSec tod)
+        in
+        MYSQL_TIME
+            { mysqlTimeYear = fromIntegral year
+            , mysqlTimeMonth = fromIntegral month
+            , mysqlTimeDay = fromIntegral dayOfMonth
+            , mysqlTimeHour = fromIntegral $ todHour tod
+            , mysqlTimeMinute = fromIntegral $ todMin tod
+            , mysqlTimeSecond = fromInteger seconds
+            , mysqlTimeNeg = False
+            , mysqlTimeSecondPart = fromInteger secondParts
+            }
+
+      defaultBind =
+         MYSQL_BIND
+            { mysqlBindBufferType = Types.Null
+            , mysqlBindBuffer = nullPtr
+            , mysqlBindBufferLength = 0
+            , mysqlBindLength = nullPtr
+            , mysqlBindIsNull = nullPtr
+            , mysqlBindIsUnsigned = 0
+            , mysqlBindError = nullPtr
+            }
+
+bindParams :: Statement -> [Value] -> IO ()
+bindParams statement args =
+    withStatement statement $ \mysqlStmt -> do
+    throwErrnoIf_ (> 0) "mysql_stmt_bind_param" $
+        withMany withValue args $ \binds ->
+        withArray binds $ mysql_stmt_bind_param mysqlStmt
+
+execute :: Statement -> IO ()
+execute statement =
+    withStatement statement $ \mysqlStmt -> do
+        throwErrnoIf_ (> 0)  "mysql_stmt_execute" $ mysql_stmt_execute mysqlStmt
+
+storeResult :: Statement -> IO ()
```

Comment:
should this name overlap with the Base.storeResult or should it be storeResultStatement